src/Matmar10/Money/Entity/Money.php

```diff
@@ -62,14 +64,45 @@ class Money implements MoneyInterface
      */
     protected $amountDisplay;
 
-    public function __construct(CurrencyInterface $currency) {
+    /**
+     * @param CurrencyInterface     $currency
+     * @param int|float|string|null $amount
+     *
+     * @return Money
+     */
+    public static function create(CurrencyInterface $currency, $amount = null)
+    {
+        return new static($currency, $amount);
+    }
+
+    /**
+     * @param CurrencyInterface     $currency
+     * @param int|float|string|null $amount
+     */
+    public function __construct(CurrencyInterface $currency, $amount = null)
```

Comment:
This seems like a great change. Can you add a test case for the various types that are now allowed to be fed into the constructor?